### ISSUE TITLE
Release 4.8.1 - RC 2 - E2E UX tests - Deployment on Kubernetes

### ISSUE THREAD

**Issue by juliamagan** [wazuh/wazuh#24455]
## End-to-End (E2E) Testing Guideline

- **Documentation:** Always consult the development documentation for the current stage tag at [this link](https://documentation-dev.wazuh.com/v4.8.1-rc2/index.html). Be careful because some of the description steps might refer to a current version in production, always navigate using the current development documention for the stage under test. Also, visit the following [pre-release package guide](https://docs.google.com/document/d/1fW8koy8P8OMvcxzIOXWGDw9qmKlju-XuRHD-Bdv3uP4/edit) to understand how to modify certain links and urls for the correct testing of the development packages.
- **Test Requirements:** Ensure your test comprehensively includes a full stack and agent/s deployment as per the Deployment requirements, detailing the machine OS, installed version, and revision.
- **Deployment Options:** While deployments can be local (using VMs, Vagrant, etc) or on the aws-dev account, opt for local deployments when feasible. For AWS access, coordinate with the DevOps team through [this link](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **External Accounts:** If tests require third-party accounts (e.g., GitHub, Azure, AWS, GCP), request the necessary access through the DevOps team [here](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **Alerts:** Every test should generate a minimum of one end-to-end alert, from the agent to the dashboard, irrespective of test type.
- **Multi-node Testing:** For multi-node wazuh-manager tests, ensure agents are connected to both workers and the master node.
- **Package Verification:** Use the pre-release package that matches the current TAG you're testing. Confirm its version and revision.
- **Filebeat Errors:** If you encounter errors with Filebeat during testing, refer to [this Slack discussion](https://wazuh-team.slack.com/archives/C03BDG0K6JC/p1672168163537809) for insights and resolutions.
- **Known Issues:** Familiarize yourself with previously reported issues in the Known Issues section. This helps in identifying already recognized errors during testing.
- **Reporting New Issues:** Any new errors discovered during testing that aren't listed under Known Issues should be reported. Assign the issue to the corresponding team (QA if unsure), add the `Release testing` objective and `Urgent` priority. Communicate these to the team and QA via the c-release Slack channel.
- **Test Conduct:** It's imperative to be thorough in your testing, offering enough detail for reviewers. Incomplete tests might necessitate a redo.
- **Documentation Feedback:** Encountering documentation gaps, unclear guidelines, or anything that disrupts the testing or UX? Open an issue, especially if it's not listed under Known Issues. Please answer the feedback section, this is a mandatory step.
- **Format:** If this is your first time doing this, refer to the format (but not necessarily the content, as it may vary) of previous E2E tests, here you have an example https://github.com/wazuh/wazuh/issues/13994.
- **Status and completion:** Change the issue status within your team project accordingly. Once you finish testing and write the conclusions, move it to Pending review and notify the @wazuh/devel-devops team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ). Beware that the reviewers might request additional information or task repetitions.
- **For reviewers:** Please move the issue to Pending final review and notify via Slack using the same thread if everything is ok, otherwise, perform an issue update with the requested changes and move it to On hold, increase the review_cycles in the  team project by one and notify the issue assignee via Slack using the same thread.

For the conclusions and the issue testing and updates, use the following legend:

**Status legend**
- 🟢 All checks passed
- 🟡 Found a known issue
- 🔴 Found a new error

## Issue delivery and completion
- **Initial delivery:** The issue's assignee must complete the testing and deliver the results by Jul 09, 2024 and notify the @wazuh/devel-devops team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ)
- **Review:** The @wazuh/devel-devops team will assign a reviewer and add it to the `review_assignee` field in the  project. The reviewer must then review the test steps and results. Ensure that all iteration cycles are completed by Jul 10, 2024 date (issue must be in `Pending final review` status) and notify the QA team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ).
- **Auditor:** The QA team must audit, validate the results, and close the issue by Jul 10, 2024.

## Deployment requirements

| Component | Installation | Type | OS |
|----------|--------------|------|----|
| Indexer | [Deployment on Kubernetes](https://documentation-dev.wazuh.com/v4.8.1-rc2/deployment-options/deploying-with-kubernetes/index.html) | Single node | RHEL 9 x86_64 |
| Server | [Deployment on Kubernetes](https://documentation-dev.wazuh.com/v4.8.1-rc2/deployment-options/deploying-with-kubernetes/index.html) | Multi node | RHEL 9 x86_64 |
| Dashboard | [Deployment on Kubernetes](https://documentation-dev.wazuh.com/v4.8.1-rc2/deployment-options/deploying-with-kubernetes/index.html) | - | RHEL 9 x86_64 |
| Agent | [Wazuh WUI one-liner deploy using FQDN and GROUP (created beforehand, don't use default)](https://documentation-dev.wazuh.com/v4.8.1-rc2/_images/deploy-new-agent-from-ui1.png) | - | Oracle Linux 9 x86_64 |

## Test description

Test [Deployment on Kubernetes](https://documentation-dev.wazuh.com/v4.8.1-rc2/deployment-options/deploying-with-kubernetes/index.html).
- Test the installation guide in detail
- Test changing the default passwords

Remember to update the version of the Docker image to point to the current v4.8.1-rc2 under tests. Example: vX.Y.Z-sv4.8.1-rc2e#

## Known issues

- https://github.com/wazuh/wazuh-docker/issues/916

## Conclusions

Summarize the errors detected (Known Issues included). Illustrate using the table below, removing current examples:

| **Status**     | **Test**    | **Failure type**    | **Notes**      |
|----------------|-------------|---------------------|----------------|
| 🟢 | Wazuh Kubernetes Deployment |  |  |
| 🟢 | Changing the Default Passwords|  |  |
| 🟢  | Dashboard Testing |  |  |
| 🟢 | Agent Deployment |  |  |

## Feedback

We value your feedback. Please provide insights on your testing experience.

- Was the testing guideline clear? Were there any ambiguities?
  - Guide was clear and easy to follow
- Did you face any challenges not covered by the guideline?
  - None
- Suggestions for improvement:
  - Just as for Wazuh indexer user, I suggest to add a test section after changing the Wazuh API user password.

## Reviewers validation

The criteria for completing this task is based on the validation of the conclusions and the test results by all reviewers.

All the checkboxes below must be marked in order to close this issue.
- [ ] @juliamagan
- [ ] @wazuh/devel-devops

**Comment by Rolly-M** [comment]
# Deployment on Kubernetes

## Environment

I ran the tests on an Ubuntu VM with minikube installed as my Kubernetes environment.

<details>
<summary>Details</summary>

```console
rolly@kube:~$ minikube status
minikube
type: Control Plane
host: Running
kubelet: Running
apiserver: Running
kubeconfig: Configured

rolly@kube:~$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.2 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.2 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

</details>

**Comment by davidcr01** [comment]
Unblocked, see https://github.com/wazuh/wazuh/issues/24518#issuecomment-2220254945

**Comment by Rolly-M** [comment]
I was provided with another cluster with the details below to continue testing:

```console
rolly@vm1:~$ kubectl get nodes -o wide
NAME                             STATUS                     ROLES    AGE   VERSION               INTERNAL-IP      EXTERNAL-IP      OS-IMAGE         KERNEL-VERSION                  CONTAINER-RUNTIME
ip-192-168-22-161.ec2.internal   Ready,SchedulingDisabled   <none>   20h   v1.30.0-eks-036c24b   192.168.22.161   44.201.226.207   Amazon Linux 2   5.10.219-208.866.amzn2.x86_64   containerd://1.7.11
ip-192-168-30-84.ec2.internal    Ready,SchedulingDisabled   <none>   20h   v1.30.0-eks-036c24b   192.168.30.84    54.210.171.150   Amazon Linux 2   5.10.219-208.866.amzn2.x86_64   containerd://1.7.11
ip-192-168-38-128.ec2.internal   Ready,SchedulingDisabled   <none>   20h   v1.30.0-eks-036c24b   192.168.38.128   3.227.231.245    Amazon Linux 2   5.10.219-208.866.amzn2.x86_64   containerd://1.7.11
ip-192-168-42-234.ec2.internal   Ready,SchedulingDisabled   <none>   20h   v1.30.0-eks-036c24b   192.168.42.234   100.29.188.8     Amazon Linux 2   5.10.219-208.866.amzn2.x86_64   containerd://1.7.11
ip-192-168-55-130.ec2.internal   Ready,SchedulingDisabled   <none>   20h   v1.30.0-eks-036c24b   192.168.55.130   3.235.51.180     Amazon Linux 2   5.10.219-208.866.amzn2.x86_64   containerd://1.7.11
```

<details>
<summary>Deploying Wazuh on Kubernetes</summary>

```console
rolly@vm1:~/wazuh-kubernetes$ kubectl apply -k envs/eks/
namespace/wazuh created
storageclass.storage.k8s.io/wazuh-storage created
configmap/dashboard-conf-46kfc92gfm created
configmap/indexer-conf-t8tdh7thct created
configmap/wazuh-conf-54bf8bh7fk created
secret/dashboard-certs-b4ffd22d6k created
secret/dashboard-cred created
secret/indexer-certs-b844mgh88h created
secret/indexer-cred created
secret/wazuh-api-cred created
secret/wazuh-authd-pass created
secret/wazuh-cluster-key created
service/dashboard created
service/indexer created
service/wazuh created
service/wazuh-cluster created
service/wazuh-indexer created
service/wazuh-workers created
deployment.apps/wazuh-dashboard created
statefulset.apps/wazuh-indexer created
statefulset.apps/wazuh-manager-master created
Warning: spec.template.spec.affinity.podAntiAffinity.preferredDuringSchedulingIgnoredDuringExecution[0].podAffinityTerm.labelSelector: a null labelSelector results in matching no pod
statefulset.apps/wazuh-manager-worker created
```

**Comment by Rolly-M** [comment]
## Changing the default Passwords

<details>
<summary>Wazuh indexer users</summary>

- Setting a new hash

```console
rolly@vm1:~/wazuh-kubernetes$ kubectl exec -it wazuh-indexer-0 -n wazuh -- /bin/bash
Defaulted container "wazuh-indexer" out of: wazuh-indexer, volume-mount-hack (init), increase-the-vm-max-map-count (init)


bash-5.2$ export JAVA_HOME=/usr/share/wazuh-indexer/jdk
bash /usr/share/wazuh-indexer/plugins/opensearch-security/tools/hash.sh
**************************************************************************
** This tool will be deprecated in the next major release of OpenSearch **
** https://github.com/opensearch-project/security/issues/1755           **
**************************************************************************
[Password:]
$2y$12$AjGREPMp**********************************************DOUSdSQY6
```

The new hash was added to the file `wazuh/indexer_stack/wazuh-indexer/indexer_conf/internal_users.yml` for the admin user

- Setting the new password
```console
rolly@vm1:~/wazuh-kubernetes$ echo -n "*********" | base64
QWJj*****zQ=

Change the admin user password in `wazuh/secrets/indexer-cred-secret.yaml`

# Wazuh API credentials secret

apiVersion: v1
kind: Secret
metadata:
  name: indexer-cred
data:
  username: YWRtaW4=              # string "admin" base64 encoded
  password: QWJj*****zQ=  # string "SecretPassword" base64 encoded
```

**Comment by Rolly-M** [comment]
## Wazuh agent deployment

```console
ubuntu@ip-172-31-31-75:~$ wget https://packages-dev.wazuh.com/pre-release/apt/pool/main/w/wazuh-agent/wazuh-agent_4.8.1-1_amd64.deb && sudo WAZUH_MANAGER='internal-ac4c4b9d1a13d4f268bc2daf20e4b491-1056834644.us-west-1.elb.amazonaws.com'
 WAZUH_REGISTRATION_PASSWORD=$'password' WAZUH_AGENT_GROUP='wazuh-EKS' WAZUH_AGENT_NAME='Agent1' dpkg -i ./wazuh-agent_4.8.1-1_amd64.deb
--2024-07-10 14:12:53--  https://packages-dev.wazuh.com/pre-release/apt/pool/main/w/wazuh-agent/wazuh-agent_4.8.1-1_amd64.deb
Resolving packages-dev.wazuh.com (packages-dev.wazuh.com)... 99.84.208.121, 99.84.208.42, 99.84.208.124, ...
Connecting to packages-dev.wazuh.com (packages-dev.wazuh.com)|99.84.208.121|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 10270908 (9.8M) [binary/octet-stream]
Saving to: ‘wazuh-agent_4.8.1-1_amd64.deb’

wazuh-agent_4.8.1-1_amd64.deb                              100%[========================================================================================================================================>]   9.79M  --.-KB/s    in 0.06s

2024-07-10 14:12:54 (152 MB/s) - ‘wazuh-agent_4.8.1-1_amd64.deb’ saved [10270908/10270908]

Selecting previously unselected package wazuh-agent.
(Reading database ... 102450 files and directories currently installed.)
Preparing to unpack .../wazuh-agent_4.8.1-1_amd64.deb ...
Unpacking wazuh-agent (4.8.1-1) ...
Setting up wazuh-agent (4.8.1-1) ...


ubuntu@ip-172-31-31-75:~$ sudo systemctl daemon-reload
sudo systemctl enable wazuh-agent
sudo systemctl start wazuh-agent
Created symlink /etc/systemd/system/multi-user.target.wants/wazuh-agent.service → /usr/lib/systemd/system/wazuh-agent.service.


ubuntu@ip-172-31-31-75:~$ sudo systemctl status wazuh-agent
● wazuh-agent.service - Wazuh agent
     Loaded: loaded (/usr/lib/systemd/system/wazuh-agent.service; enabled; preset: enabled)
     Active: active (running) since Wed 2024-07-10 14:15:17 UTC; 1min 8s ago
    Process: 20401 ExecStart=/usr/bin/env /var/ossec/bin/wazuh-control start (code=exited, status=0/SUCCESS)
      Tasks: 27 (limit: 1130)
     Memory: 16.0M (peak: 19.0M)
        CPU: 334ms
     CGroup: /system.slice/wazuh-agent.service
             ├─20447 /var/ossec/bin/wazuh-execd
             ├─20565 /var/ossec/bin/wazuh-agentd
             ├─20755 /var/ossec/bin/wazuh-syscheckd
             ├─20776 /var/ossec/bin/wazuh-logcollector
             └─20816 /var/ossec/bin/wazuh-modulesd

Jul 10 14:15:10 ip-172-31-31-75 systemd[1]: Starting wazuh-agent.service - Wazuh agent...
Jul 10 14:15:10 ip-172-31-31-75 env[20401]: Starting Wazuh v4.8.1...
Jul 10 14:15:11 ip-172-31-31-75 env[20401]: Started wazuh-execd...
Jul 10 14:15:12 ip-172-31-31-75 env[20401]: Started wazuh-agentd...
Jul 10 14:15:13 ip-172-31-31-75 env[20401]: Started wazuh-syscheckd...
Jul 10 14:15:14 ip-172-31-31-75 env[20401]: Started wazuh-logcollector...
Jul 10 14:15:15 ip-172-31-31-75 env[20401]: Started wazuh-modulesd...
Jul 10 14:15:17 ip-172-31-31-75 env[20401]: Completed.
Jul 10 14:15:17 ip-172-31-31-75 systemd[1]: Started wazuh-agent.service - Wazuh agent.
```

Agent was successfully enrolled to the Wazuh manager.
